source/Background/Controller.js

```diff
@@ -474,4 +474,86 @@ backgroundController.exposeController(
   },
 );
 
+backgroundController.exposeController(
+  'requestBurnXTC',
+  async (opts, metadata, args) => secureController(opts.callback, async () => {
+    const { message, sender, callback } = opts;
+
+    const { id: callId } = message.data.data;
+    const { id: portId } = sender;
+    storage.get(keyring.currentWalletId.toString(), async (state) => {
+      const apps = state?.[keyring.currentWalletId]?.apps || {};
+      if (
+        apps?.[metadata.url]?.status === CONNECTION_STATUS.accepted
+      ) {
+        const argsError = validateBurnArgs(args);
+        if (argsError) {
+          callback(argsError, null);
+          return;
+        }
+        const url = qs.stringifyUrl({
+          url: 'notification.html',
+          query: {
+            callId,
+            portId,
+            metadataJson: JSON.stringify(metadata),
+            argsJson: JSON.stringify(args),
+            type: 'burnXTC',
+          },
+        });
+
+        const height = keyring?.isUnlocked
+          ? SIZES.detailHeightSmall
+          : SIZES.loginHeight;
+        extension.windows.create({
+          url,
+          type: 'popup',
+          width: SIZES.width,
+          height,
+          top: 65,
+          left: metadata.pageWidth - SIZES.width,
+        });
+      } else {
+        callback(ERRORS.CONNECTION_ERROR, null);
+      }
+    });
+  }),
+);
+
+backgroundController.exposeController(
+  'handleRequestBurnXTC',
+  async (opts, transferRequests, callId, portId) => {
+    const { callback } = opts;
+    const transfer = transferRequests?.[0];
+
+    // Answer this callback no matter if the transfer succeeds or not.
+    if (transfer?.status === 'declined') {
+      callback(null, true);
+      callback(ERRORS.TRANSACTION_REJECTED, null, [{ portId, callId }]);
+    } else {
+      const burnXTC = getKeyringHandler(HANDLER_TYPES.BURN_XTC, keyring);
+      const getBalance = getKeyringHandler(HANDLER_TYPES.GET_BALANCE, keyring);
+      const assets = await getBalance();
+
+      if (assets?.[1]?.amount * CYCLES_PER_TC > transfer.amount) {
```

Comment:
this [1] is kinda monkaW, but i assume its XTC

Reply:
Yep, it refers to XTC but we should probably refactor it

Reply:
you guys should've changed this :topkek: